### PR TITLE
Flow fixes ( Working***) 

### PR DESCRIPTION
**Pull request by mattanger**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

- **Other information**:

### CODE REVIEW

File: src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue [hunk 1]
```diff
@@ -224,13 +227,17 @@ export default {
...
+      const gageInfoPromise = this.fetchGaugeInfo()
       this.$store.dispatch(
         readingsActions.FETCH_GAGE_READINGS_DATA,
         this.formData
       )
+      await gageInfoPromise
+    },
+    async fetchGaugeInfo () {
+      this.$store.dispatch(gaugeInfoActions.FETCH_GAGE_INFO, this.formData.gauge_id)
     }
```
Comment (drewalth) [inline src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue:241]: This seems odd. Why not just call the dispatch directly on L37? `await this.$store.dispatch`

**Review by drewalth — APPROVED** [review]

**Comment by drewalth** [comment]
@mattanger what ticket is this for?

**Comment by mattanger** [comment]
This is just a working branch where I was messing with fixing how flows were pulled, gauge metrics weren't being used correctly and/or pulled. Kind of just a code dump at the moment.

File: src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue (hunk 1 above)
Comment (mattanger) [reply src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue:241]: yeah I though that was going to be a call that was going to be used more than once, and I don't really like having `await this.$store.dispatch` sprinkled all over.

Comment (drewalth) [reply src/app/views/river-detail/flow-tab/components/gage-chart-controls.vue:241]: @mattanger True that. `mapActions` might help. 

```js
methods: {
  ...mapActions({
    loadGauge: 'GaugeInfoSubModule/FETCH_GAGE_INFO'
  })
}
```

Cant remember what the submodule is called. or how crazy nested it is... 

Then you could just call:

```js
this.loadGauge(this.formData.gauge_id)
```

A little nicer.

**Pull request closed without merging**